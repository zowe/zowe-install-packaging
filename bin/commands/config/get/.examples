zwe config get -p path-of-configuration -c /path/to/zowe.yaml
zwe config get -p .zowe.certificate -c /path/to/zowe.yaml
zwe config get -p .zowe.components[] -c /path/to/zowe.yaml
zwe config get -p .zowe.components.gateway -c /path/to/zowe.yaml
zwe config get -p .zowe.components.gateway.enabled -c /path/to/zowe.yaml