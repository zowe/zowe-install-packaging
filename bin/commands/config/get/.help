Return value of a configuration defined in YAML configuration.
This command requires zowe.useConfigmgr=true or --configmgr to be used.
