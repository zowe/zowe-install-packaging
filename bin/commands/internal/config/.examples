    zwe internal config get -p path-of-configuration -c /path/to/zowe.yaml

    zwe internal config set -p path-of-configuration -e new-value-of-configuration -c /path/to/zowe.yaml
    