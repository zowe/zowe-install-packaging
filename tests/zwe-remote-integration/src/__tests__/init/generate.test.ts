/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { REMOTE_SYSTEM_INFO } from '../../config/TestConfig';
import ZoweYamlType from '../../config/ZoweYamlType';
import { RemoteTestRunner } from '../../zos/RemoteTestRunner';
import { ZoweConfig } from '../../config/ZoweConfig';
import { FileType, TestFileActions, TestFile } from '../../zos/TestFileActions';

const testSuiteName = 'init-generate';
describe(`${testSuiteName}`, () => {
  let testRunner: RemoteTestRunner;
  let cfgYaml: ZoweYamlType;
  let defaultCfgYaml: ZoweYamlType;
  let cleanupDatasets: TestFile[] = []; // a list of datasets deleted after every test

  beforeAll(() => {
    testRunner = new RemoteTestRunner(testSuiteName);
  });
  beforeEach(() => {
    cfgYaml = ZoweConfig.getZoweYaml();
    defaultCfgYaml = ZoweConfig.getDefaultsYaml();
  });

  afterEach(async () => {
    await testRunner.postTest();

    // re-created in every `init generate` based on changes to zowe yaml command...
    const jcllib: TestFile = { name: REMOTE_SYSTEM_INFO.jcllib, type: FileType.DS_NON_CLUSTER };

    // try to delete everything we know about
    await TestFileActions.deleteAll([...cleanupDatasets, jcllib]);
    cleanupDatasets = [];
  });

  afterAll(() => {
    testRunner.shutdown();
  });

  describe('(SHORT)', () => {
    it('disable cfgmgr', async () => {
      cfgYaml.zowe.useConfigmgr = false;
      const result = await testRunner.runZweTest(cfgYaml, 'init generate');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(60); // 60 is expected...
    });

    it('bad ds prefix', async () => {
      cfgYaml.zowe.setup.dataset.prefix = 'ZOWEAD3.ZWETEST.NOEXIST';
      const result = await testRunner.runZweTest(cfgYaml, 'init generate --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(143);
    });

    it('simple --dry-run', async () => {
      const result = await testRunner.runZweTest(cfgYaml, 'init generate --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(0); // 60 is expected...
    });

    it('BAD: missing defaults.yaml', async () => {
      await testRunner.removeFileForTest('files/defaults.yaml');
      const result = await testRunner.runZweTest(cfgYaml, 'init vsam --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot(); // FIXME: the snapshot indicates processing continues when it shouldn't
      expect(result.rc).not.toBe(0);
    });

    it('BAD: invalid value defaults.yaml', async () => {
      // @ts-expect-error intentionally setting an incorrect value
      defaultCfgYaml.zowe.configmgr.validation = 'WRONG_VALUE';
      const result = await testRunner.runZweTest(cfgYaml, 'init vsam --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot(); // FIXME: the snapshot indicates processing continues when it shouldn't
      expect(result.rc).not.toBe(0);
    });

    it('BAD: invalid format defaults.yaml', async () => {
      // @ts-expect-error invalid yaml format
      defaultCfgYaml.zowe = '....\n somefield:\n  #another:\n' + defaultCfgYaml.zowe;
      const result = await testRunner.runZweTest(cfgYaml, 'init vsam --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot(); // FIXME: the snapshot indicates processing continues when it shouldn't
      expect(result.rc).not.toBe(0);
    });
  });

  describe('(LONG)', () => {
    it('missing proclib with valid stcs abc', async () => {
      cfgYaml.zowe.setup.dataset.proclib = `${REMOTE_SYSTEM_INFO.prefix}.NOEXIST.PROC`;
      const result = await testRunner.runZweTest(cfgYaml, 'init stc --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(1);
    });
  });
});
