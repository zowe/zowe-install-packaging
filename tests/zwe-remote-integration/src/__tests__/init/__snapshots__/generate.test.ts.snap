// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init-generate (SHORT) BAD: invalid format defaults.yaml 1`] = `
"
$ Pre-load
Load status: 0
Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
-------------------------------------------------------------------------------
>> Initialize Zowe custom data sets

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEGENER)
--- JCL content ---
//ZWEGENER JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* This job is responsible for generating other jobs required
//* to configure Zowe.
//*
//* The method of validating your configuration is using
//* JSON Schema <https://json-schema.org>. Zowe provides
//* the ConfigMgr to assist in this. This job will invoke
//* the ConfigMgr to validate your current configuration
//* before generating any jobs. If there are any values
//* that are incorrect, you will be notified. You should
//* fix the value and then run this job again. You can run
//* this job as many times as you need.
//*
//* Configmgr documentation:
//* https://docs.zowe.org/stable/user-guide/configmgr-using
//*
//* Note: Any string with braces has an associated yaml value
//*       in one of the yaml definitions for Zowe.
//*       You must find the value and substitute it.
//*
//*       {key} -> value
//*
//GENER    EXEC PGM=IKJEFT1B
//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
//   SPACE=(3120,(20,5,10))
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//SYSPROC  DD DSN=TEST.DATASET.PFX.SZWEEXEC,DISP=SHR
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//STEPLIB  DD DSN=TEST.DATASET.PFX.SZWELOAD,DISP=SHR
//ISPPLIB  DD DSN=ISP.SISPPENU,DISP=SHR
//ISPMLIB  DD DSN=ISP.SISPMENU,DISP=SHR
//ISPTLIB  DD DSN=ISP.SISPTENU,DISP=SHR
//ISPSLIB  DD DSN=ISP.SISPSENU,DISP=SHR
//*
//* The order must be as follows.
//*
//* zowe-yaml-schema.json
//* server-common.json
//*
//* Replace /test/dir with where your Zowe run time
//* directory is, as seen in zowe.yaml
//*
//MYSCHEMA DD   *,DLM=$$
FILE /test/dir/schemas/zowe-yaml-schema.json
FILE /test/dir/schemas/server-common.json
$$
//*
//* The DD below must include one or more FILE or PARMLIB
//* Entries. The lower entries have their values
//* Overridden by the higher entries.
//* PARMLIB member must be named "ZWEYAML"
//*
//* Do not remove the defaults.yaml entry.
//*
//* Ex. PARMLIB MY.ZOWE.CUSTOM.PARMLIB
//*     FILE /the/zowe/defaults.yaml
//MYCONFIG DD   *,DLM=$$
FILE /test/dir/zowe.test.yaml
FILE /test/dir/files/defaults.yaml
$$
//CMGROUT  DD   SYSOUT=*
//SYSPRINT DD   SYSOUT=*
//SYSTSPRT DD   SYSOUT=*
//*
//* Change 'generate' to 'nogenerate' if you only
//* want to validate your configuration. The default
//* option, 'generate', will validate and then generate
//* jobs based on your configuration.
//*
//*   - generate
//*   - nogenerate
//*
//* Change 'noverbose' to 'verbose' below for
//* advanced logging. This is not needed unless
//* there is an error.
//*
//*   - verbose
//*   - noverbose
//*
//SYSTSIN  DD   *
ISPSTART CMD(%ZWEGEN00 -
generate -
noverbose -
)

--- End of JCL ---
Submitting Job ZWEGENER
Job ZWEGENER(JOB00000) completed with RC=0
Zowe JCL generated successfully
Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWECSVSM) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWECSVSM)
--- JCL Content ---
//ZWECSVSM JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* Zowe Open Source Project
//* This JCL creates the VSAM data set for the Caching Service.
//*
//*
//* CAUTION: This is neither a JCL procedure nor a complete job.
//* Before using this JCL, you will have to make the following
//* modifications:
//*
//* 1) Add job name and job parameters to the JOB statement, to
//*    meet your system requirements.
//*
//* When using RLS, customize the definitions in the RLS DD:
//*
//* 2) Optionally, change LOG option NONE to UNDO or ALL to set the
//*    desired recovery options for the RLS VSAM.
//*
//*
//* Note(s):
//*
//* 1. This job should complete with return code 0.
//*
//********************************************************************
//*
//ALLOC    EXEC PGM=IDCAMS,REGION=0M
//SYSPRINT DD SYSOUT=*
//SYSIN    DD *
  DEFINE CLUSTER -
   (NAME(TEST.DATASET.PFX.VSAM) -
//         DD DDNAME=NONRLS
//         DD *
    REC(80 20) -
    INDEXED) -
   DATA(NAME(TEST.DATASET.PFX.VSAM.DATA) -
    RECSZ(4096 4096) -
    UNIQUE -
    KEYS(128 0)) -
   INDEX(NAME(TEST.DATASET.PFX.VSAM.INDEX) -
    UNIQUE)
//RLS      DD *
    STORCLAS() -
    LOG(NONE) -
//NONRLS   DD *
    VOLUME(TSTVOL) -
    SHAREOPTIONS(2 3) -
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe Caching Service VSAM storage is created successfully.
"
`;

exports[`init-generate (SHORT) BAD: invalid value defaults.yaml 1`] = `
"
$ Pre-load
Load status: 0
Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
-------------------------------------------------------------------------------
>> Initialize Zowe custom data sets

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEGENER)
--- JCL content ---
//ZWEGENER JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* This job is responsible for generating other jobs required
//* to configure Zowe.
//*
//* The method of validating your configuration is using
//* JSON Schema <https://json-schema.org>. Zowe provides
//* the ConfigMgr to assist in this. This job will invoke
//* the ConfigMgr to validate your current configuration
//* before generating any jobs. If there are any values
//* that are incorrect, you will be notified. You should
//* fix the value and then run this job again. You can run
//* this job as many times as you need.
//*
//* Configmgr documentation:
//* https://docs.zowe.org/stable/user-guide/configmgr-using
//*
//* Note: Any string with braces has an associated yaml value
//*       in one of the yaml definitions for Zowe.
//*       You must find the value and substitute it.
//*
//*       {key} -> value
//*
//GENER    EXEC PGM=IKJEFT1B
//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
//   SPACE=(3120,(20,5,10))
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//SYSPROC  DD DSN=TEST.DATASET.PFX.SZWEEXEC,DISP=SHR
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//STEPLIB  DD DSN=TEST.DATASET.PFX.SZWELOAD,DISP=SHR
//ISPPLIB  DD DSN=ISP.SISPPENU,DISP=SHR
//ISPMLIB  DD DSN=ISP.SISPMENU,DISP=SHR
//ISPTLIB  DD DSN=ISP.SISPTENU,DISP=SHR
//ISPSLIB  DD DSN=ISP.SISPSENU,DISP=SHR
//*
//* The order must be as follows.
//*
//* zowe-yaml-schema.json
//* server-common.json
//*
//* Replace /test/dir with where your Zowe run time
//* directory is, as seen in zowe.yaml
//*
//MYSCHEMA DD   *,DLM=$$
FILE /test/dir/schemas/zowe-yaml-schema.json
FILE /test/dir/schemas/server-common.json
$$
//*
//* The DD below must include one or more FILE or PARMLIB
//* Entries. The lower entries have their values
//* Overridden by the higher entries.
//* PARMLIB member must be named "ZWEYAML"
//*
//* Do not remove the defaults.yaml entry.
//*
//* Ex. PARMLIB MY.ZOWE.CUSTOM.PARMLIB
//*     FILE /the/zowe/defaults.yaml
//MYCONFIG DD   *,DLM=$$
FILE /test/dir/zowe.test.yaml
FILE /test/dir/files/defaults.yaml
$$
//CMGROUT  DD   SYSOUT=*
//SYSPRINT DD   SYSOUT=*
//SYSTSPRT DD   SYSOUT=*
//*
//* Change 'generate' to 'nogenerate' if you only
//* want to validate your configuration. The default
//* option, 'generate', will validate and then generate
//* jobs based on your configuration.
//*
//*   - generate
//*   - nogenerate
//*
//* Change 'noverbose' to 'verbose' below for
//* advanced logging. This is not needed unless
//* there is an error.
//*
//*   - verbose
//*   - noverbose
//*
//SYSTSIN  DD   *
ISPSTART CMD(%ZWEGEN00 -
generate -
noverbose -
)

--- End of JCL ---
Submitting Job ZWEGENER
Job ZWEGENER(JOB00000) completed with RC=0
Zowe JCL generated successfully
Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWECSVSM) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWECSVSM)
--- JCL Content ---
//ZWECSVSM JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* Zowe Open Source Project
//* This JCL creates the VSAM data set for the Caching Service.
//*
//*
//* CAUTION: This is neither a JCL procedure nor a complete job.
//* Before using this JCL, you will have to make the following
//* modifications:
//*
//* 1) Add job name and job parameters to the JOB statement, to
//*    meet your system requirements.
//*
//* When using RLS, customize the definitions in the RLS DD:
//*
//* 2) Optionally, change LOG option NONE to UNDO or ALL to set the
//*    desired recovery options for the RLS VSAM.
//*
//*
//* Note(s):
//*
//* 1. This job should complete with return code 0.
//*
//********************************************************************
//*
//ALLOC    EXEC PGM=IDCAMS,REGION=0M
//SYSPRINT DD SYSOUT=*
//SYSIN    DD *
  DEFINE CLUSTER -
   (NAME(TEST.DATASET.PFX.VSAM) -
//         DD DDNAME=NONRLS
//         DD *
    REC(80 20) -
    INDEXED) -
   DATA(NAME(TEST.DATASET.PFX.VSAM.DATA) -
    RECSZ(4096 4096) -
    UNIQUE -
    KEYS(128 0)) -
   INDEX(NAME(TEST.DATASET.PFX.VSAM.INDEX) -
    UNIQUE)
//RLS      DD *
    STORCLAS() -
    LOG(NONE) -
//NONRLS   DD *
    VOLUME(TSTVOL) -
    SHAREOPTIONS(2 3) -
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe Caching Service VSAM storage is created successfully.
"
`;

exports[`init-generate (SHORT) BAD: missing defaults.yaml 1`] = `
"
$ Pre-load
ZWEL0318E - failed to read '/test/dir/files/defaults.yaml' - EDC5129I No such file or directory. (errno2=0x05620062)
Load status: 0
Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
-------------------------------------------------------------------------------
>> Initialize Zowe custom data sets

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEGENER)
--- JCL content ---
//ZWEGENER JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* This job is responsible for generating other jobs required
//* to configure Zowe.
//*
//* The method of validating your configuration is using
//* JSON Schema <https://json-schema.org>. Zowe provides
//* the ConfigMgr to assist in this. This job will invoke
//* the ConfigMgr to validate your current configuration
//* before generating any jobs. If there are any values
//* that are incorrect, you will be notified. You should
//* fix the value and then run this job again. You can run
//* this job as many times as you need.
//*
//* Configmgr documentation:
//* https://docs.zowe.org/stable/user-guide/configmgr-using
//*
//* Note: Any string with braces has an associated yaml value
//*       in one of the yaml definitions for Zowe.
//*       You must find the value and substitute it.
//*
//*       {key} -> value
//*
//GENER    EXEC PGM=IKJEFT1B
//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
//   SPACE=(3120,(20,5,10))
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//SYSPROC  DD DSN=TEST.DATASET.PFX.SZWEEXEC,DISP=SHR
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//STEPLIB  DD DSN=TEST.DATASET.PFX.SZWELOAD,DISP=SHR
//ISPPLIB  DD DSN=ISP.SISPPENU,DISP=SHR
//ISPMLIB  DD DSN=ISP.SISPMENU,DISP=SHR
//ISPTLIB  DD DSN=ISP.SISPTENU,DISP=SHR
//ISPSLIB  DD DSN=ISP.SISPSENU,DISP=SHR
//*
//* The order must be as follows.
//*
//* zowe-yaml-schema.json
//* server-common.json
//*
//* Replace /test/dir with where your Zowe run time
//* directory is, as seen in zowe.yaml
//*
//MYSCHEMA DD   *,DLM=$$
FILE /test/dir/schemas/zowe-yaml-schema.json
FILE /test/dir/schemas/server-common.json
$$
//*
//* The DD below must include one or more FILE or PARMLIB
//* Entries. The lower entries have their values
//* Overridden by the higher entries.
//* PARMLIB member must be named "ZWEYAML"
//*
//* Do not remove the defaults.yaml entry.
//*
//* Ex. PARMLIB MY.ZOWE.CUSTOM.PARMLIB
//*     FILE /the/zowe/defaults.yaml
//MYCONFIG DD   *,DLM=$$
FILE /test/dir/zowe.test.yaml
FILE /test/dir/files/defaults.yaml
$$
//CMGROUT  DD   SYSOUT=*
//SYSPRINT DD   SYSOUT=*
//SYSTSPRT DD   SYSOUT=*
//*
//* Change 'generate' to 'nogenerate' if you only
//* want to validate your configuration. The default
//* option, 'generate', will validate and then generate
//* jobs based on your configuration.
//*
//*   - generate
//*   - nogenerate
//*
//* Change 'noverbose' to 'verbose' below for
//* advanced logging. This is not needed unless
//* there is an error.
//*
//*   - verbose
//*   - noverbose
//*
//SYSTSIN  DD   *
ISPSTART CMD(%ZWEGEN00 -
generate -
noverbose -
)

--- End of JCL ---
Submitting Job ZWEGENER
Job ZWEGENER(JOB00000) completed with RC=0
Zowe JCL generated successfully
Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWECSVSM) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWECSVSM)
--- JCL Content ---
//ZWECSVSM JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* Zowe Open Source Project
//* This JCL creates the VSAM data set for the Caching Service.
//*
//*
//* CAUTION: This is neither a JCL procedure nor a complete job.
//* Before using this JCL, you will have to make the following
//* modifications:
//*
//* 1) Add job name and job parameters to the JOB statement, to
//*    meet your system requirements.
//*
//* When using RLS, customize the definitions in the RLS DD:
//*
//* 2) Optionally, change LOG option NONE to UNDO or ALL to set the
//*    desired recovery options for the RLS VSAM.
//*
//*
//* Note(s):
//*
//* 1. This job should complete with return code 0.
//*
//********************************************************************
//*
//ALLOC    EXEC PGM=IDCAMS,REGION=0M
//SYSPRINT DD SYSOUT=*
//SYSIN    DD *
  DEFINE CLUSTER -
   (NAME(TEST.DATASET.PFX.VSAM) -
//         DD DDNAME=NONRLS
//         DD *
    REC(80 20) -
    INDEXED) -
   DATA(NAME(TEST.DATASET.PFX.VSAM.DATA) -
    RECSZ(4096 4096) -
    UNIQUE -
    KEYS(128 0)) -
   INDEX(NAME(TEST.DATASET.PFX.VSAM.INDEX) -
    UNIQUE)
//RLS      DD *
    STORCLAS() -
    LOG(NONE) -
//NONRLS   DD *
    VOLUME(TSTVOL) -
    SHAREOPTIONS(2 3) -
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe Caching Service VSAM storage is created successfully.
"
`;

exports[`init-generate (SHORT) bad ds prefix 1`] = `
"
$ Pre-load
Load status: 0
Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
cp: FSUM6258 cannot open file "//'TEST.DATASET.PFX.NOEXIST.SZWESAMP(ZWEGENER)'": EDC5049I The specified file name could not be located. (errno2=0xC00B0641)
ERROR: ZWEL0143E Cannot find data set member 'TEST.DATASET.PFX.NOEXIST.SZWESAMP(ZWEGENER)'. You may need to re-run zwe install."
`;

exports[`init-generate (SHORT) disable cfgmgr 1`] = `
"
$ Pre-load
Load status: 0
Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEGENER)
--- JCL content ---
//ZWEGENER JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* This job is responsible for generating other jobs required
//* to configure Zowe.
//*
//* The method of validating your configuration is using
//* JSON Schema <https://json-schema.org>. Zowe provides
//* the ConfigMgr to assist in this. This job will invoke
//* the ConfigMgr to validate your current configuration
//* before generating any jobs. If there are any values
//* that are incorrect, you will be notified. You should
//* fix the value and then run this job again. You can run
//* this job as many times as you need.
//*
//* Configmgr documentation:
//* https://docs.zowe.org/stable/user-guide/configmgr-using
//*
//* Note: Any string with braces has an associated yaml value
//*       in one of the yaml definitions for Zowe.
//*       You must find the value and substitute it.
//*
//*       {key} -> value
//*
//GENER    EXEC PGM=IKJEFT1B
//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
//   SPACE=(3120,(20,5,10))
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//SYSPROC  DD DSN=TEST.DATASET.PFX.SZWEEXEC,DISP=SHR
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//STEPLIB  DD DSN=TEST.DATASET.PFX.SZWELOAD,DISP=SHR
//ISPPLIB  DD DSN=ISP.SISPPENU,DISP=SHR
//ISPMLIB  DD DSN=ISP.SISPMENU,DISP=SHR
//ISPTLIB  DD DSN=ISP.SISPTENU,DISP=SHR
//ISPSLIB  DD DSN=ISP.SISPSENU,DISP=SHR
//*
//* The order must be as follows.
//*
//* zowe-yaml-schema.json
//* server-common.json
//*
//* Replace /test/dir with where your Zowe run time
//* directory is, as seen in zowe.yaml
//*
//MYSCHEMA DD   *,DLM=$$
FILE /test/dir/schemas/zowe-yaml-schema.json
FILE /test/dir/schemas/server-common.json
$$
//*
//* The DD below must include one or more FILE or PARMLIB
//* Entries. The lower entries have their values
//* Overridden by the higher entries.
//* PARMLIB member must be named "ZWEYAML"
//*
//* Do not remove the defaults.yaml entry.
//*
//* Ex. PARMLIB MY.ZOWE.CUSTOM.PARMLIB
//*     FILE /the/zowe/defaults.yaml
//MYCONFIG DD   *,DLM=$$
FILE /test/dir/zowe.test.yaml
FILE /test/dir/files/defaults.yaml
$$
//CMGROUT  DD   SYSOUT=*
//SYSPRINT DD   SYSOUT=*
//SYSTSPRT DD   SYSOUT=*
//*
//* Change 'generate' to 'nogenerate' if you only
//* want to validate your configuration. The default
//* option, 'generate', will validate and then generate
//* jobs based on your configuration.
//*
//*   - generate
//*   - nogenerate
//*
//* Change 'noverbose' to 'verbose' below for
//* advanced logging. This is not needed unless
//* there is an error.
//*
//*   - verbose
//*   - noverbose
//*
//SYSTSIN  DD   *
ISPSTART CMD(%ZWEGEN00 -
generate -
noverbose -
)

--- End of JCL ---
Submitting Job ZWEGENER
Job ZWEGENER(JOB00000) completed with RC=0
Zowe JCL generated successfully"
`;

exports[`init-generate (SHORT) simple --dry-run 1`] = `
"
$ Pre-load
Load status: 0
Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEGENER)
--- JCL content ---
//ZWEGENER JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* This job is responsible for generating other jobs required
//* to configure Zowe.
//*
//* The method of validating your configuration is using
//* JSON Schema <https://json-schema.org>. Zowe provides
//* the ConfigMgr to assist in this. This job will invoke
//* the ConfigMgr to validate your current configuration
//* before generating any jobs. If there are any values
//* that are incorrect, you will be notified. You should
//* fix the value and then run this job again. You can run
//* this job as many times as you need.
//*
//* Configmgr documentation:
//* https://docs.zowe.org/stable/user-guide/configmgr-using
//*
//* Note: Any string with braces has an associated yaml value
//*       in one of the yaml definitions for Zowe.
//*       You must find the value and substitute it.
//*
//*       {key} -> value
//*
//GENER    EXEC PGM=IKJEFT1B
//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
//   SPACE=(3120,(20,5,10))
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//SYSPROC  DD DSN=TEST.DATASET.PFX.SZWEEXEC,DISP=SHR
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//STEPLIB  DD DSN=TEST.DATASET.PFX.SZWELOAD,DISP=SHR
//ISPPLIB  DD DSN=ISP.SISPPENU,DISP=SHR
//ISPMLIB  DD DSN=ISP.SISPMENU,DISP=SHR
//ISPTLIB  DD DSN=ISP.SISPTENU,DISP=SHR
//ISPSLIB  DD DSN=ISP.SISPSENU,DISP=SHR
//*
//* The order must be as follows.
//*
//* zowe-yaml-schema.json
//* server-common.json
//*
//* Replace /test/dir with where your Zowe run time
//* directory is, as seen in zowe.yaml
//*
//MYSCHEMA DD   *,DLM=$$
FILE /test/dir/schemas/zowe-yaml-schema.json
FILE /test/dir/schemas/server-common.json
$$
//*
//* The DD below must include one or more FILE or PARMLIB
//* Entries. The lower entries have their values
//* Overridden by the higher entries.
//* PARMLIB member must be named "ZWEYAML"
//*
//* Do not remove the defaults.yaml entry.
//*
//* Ex. PARMLIB MY.ZOWE.CUSTOM.PARMLIB
//*     FILE /the/zowe/defaults.yaml
//MYCONFIG DD   *,DLM=$$
FILE /test/dir/zowe.test.yaml
FILE /test/dir/files/defaults.yaml
$$
//CMGROUT  DD   SYSOUT=*
//SYSPRINT DD   SYSOUT=*
//SYSTSPRT DD   SYSOUT=*
//*
//* Change 'generate' to 'nogenerate' if you only
//* want to validate your configuration. The default
//* option, 'generate', will validate and then generate
//* jobs based on your configuration.
//*
//*   - generate
//*   - nogenerate
//*
//* Change 'noverbose' to 'verbose' below for
//* advanced logging. This is not needed unless
//* there is an error.
//*
//*   - verbose
//*   - noverbose
//*
//SYSTSIN  DD   *
ISPSTART CMD(%ZWEGEN00 -
generate -
noverbose -
)

--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly."
`;
